src/harness/reference_models/fss_purge/fss_purge.py

```diff
@@ -0,0 +1,332 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+==================================================================================
+    This is a subset of the Pre-IAP reference model which implements
+    the FSS purge list algorithm described in R2-SGN-29.
+    The main function is fssPurgeModel().
+==================================================================================
+"""
+import numpy as np
+
+from collections import namedtuple
+from reference_models.antenna import antenna
+from reference_models.geo import terrain, vincenty
+from reference_models.propagation import wf_itm
+
+# Definition of constants and data types for FSS purge model.
+# Frequency used in propagation model (in MHz) [R2-SGN-04]
+FREQ_PROP_MODEL = 3625.0
+
+# Reference Channel bandwidth for FSS OOBE calculation
+REF_BW = 5.e6
+
+# One Mega Hertz
+ONE_MHZ = 1.e6
+
+# Define FSS information tuple containing the parameters necessary for antenna
+# and propagation calculation as listed in the test spec
+FssProtectionPoint = namedtuple('FssProtectionPoint',

```

Comment:
This looks like it was copied from another location. Should we extract that into a common library instead of copying?

Reply:
This   tuple is also used in the IAP reference model and is define din the   'interference' model. Interference model is already made a common library for   the 'aggregate_interfernce' and 'IAP' models. Hence we also think it makes   sense to use this tuple definition from the interference model rather than   defining a local copy here.